django_mongodb_backend/indexes.py

```diff
@@ -101,6 +105,185 @@ def where_node_idx(self, compiler, connection):
     return mql
 
 
+class SearchIndex(Index):
+    suffix = "six"
+    _error_id_prefix = "django_mongodb_backend.indexes.SearchIndex"
+
+    def check(self, model, connection):
+        errors = []
+        if not connection.features.supports_atlas_search:
+            errors.append(
+                Warning(
+                    "This MongoDB server does not support atlas search.",
+                    hint=(
+                        "The index won't be created. Silence this warning if you "
+                        "don't care about it."
+                    ),
+                    obj=self,
```

Comment:
While working on [the hook in Django](https://github.com/django/django/pull/19395), I noticed the built-in checks provide `obj=model` which may be more helpful for locating the problematic index. But perhaps the warning message would also be more clear if it said: ``f"This MongoDB server does not support {self.__class__.__name__}."`